src/Features/Core/Portable/SolutionCrawler/WorkCoordinator.NormalPriorityProcessor.cs

```diff
@@ -335,7 +335,8 @@ private async Task ProcessDocumentAsync(ImmutableArray<IIncrementalAnalyzer> ana
                             {
                                 var document = solution.GetDocument(documentId);
 
-                                if (document != null)
+                                if (document != null &&
+                                    !workItem.InvocationReasons.Contains(PredefinedInvocationReasons.DocumentRemoved))
```

Comment:
i don't like the asymettry of this check with the previous code i saw.  why did the previous code care about 'closed or removed' but htis code only cares about 'removed'?  We need to doc these carefully to make it clear why these are the right semantics.

Reply:
Ah, this was just left behind from a different approach I tried for this. I will revert this change.